src/ui/pages/SwapForm/Quotes/RateLine.tsx

```diff
@@ -91,79 +130,143 @@ export function RateLine({
-                      alt={`${selectedQuote.contract_metadata?.name} logo`}
+                {selectedQuote.contract_metadata?.icon_url ? (
+                  <div
+                    style={{
+                      position: 'relative',
+                      width: 20,
+                      height: 20,
+                      // overflow: 'hidden',
+                    }}
+                  >
+                    <SlidingRectangle
+                      size={20}
+                      src={selectedQuote.contract_metadata.icon_url}
+                      render={(src, index) => (
+                        <img
+                          title={selectedQuote.contract_metadata?.name}
+                          style={{
+                            position: 'absolute',
+                            left: 0,
+                            right: 0,
+                            width: 20,
+                            height: 20,
+                            borderRadius: protocolBorderRadius,
+                            zIndex: index,
+                          }}
+                          src={src}
+                          // The alt here may be from a sibling image, but hopefully it doesn't matter
+                          alt={`${selectedQuote.contract_metadata?.name} logo`}
+                        />
+                      )}
                     />
-                  )}
-                />
-              </div>
-            ) : null}
-            {rate ? (
-              `1 ${rate.leftAsset.symbol} = ${formatTokenValue(
-                rate.value,
-                rate.rightAsset.symbol
-              )}`
-            ) : (
-              <span>{selectedQuote.contract_metadata?.name ?? 'unknown'}</span>
-            )}
-          </HStack>
-        ) : error ? (
-          <UIText kind="small/regular" color="var(--notice-600)">
-            {getQuotesErrorMessage(quotesData)}
-          </UIText>
-        ) : null}
-      </span>
-    </HStack>
+                  </div>
+                ) : null}
+                {rate ? (
+                  <UIText kind="small/accent" color="var(--primary)">
+                    {`1 ${rate.leftAsset.symbol} = ${formatTokenValue(
+                      rate.value,
+                      rate.rightAsset.symbol
+                    )}`}
+                  </UIText>
+                ) : (
+                  <span>
+                    {selectedQuote.contract_metadata?.name ?? 'unknown'}
+                  </span>
+                )}
+              </HStack>
+            </UnstyledButton>
+          ) : error ? (
+            <UIText kind="small/regular" color="var(--notice-600)">
+              {getQuotesErrorMessage(quotesData)}
+            </UIText>
+          ) : null}
+        </span>
+      </HStack>
+      {userFeeTier && selectedQuote ? (
+        <BottomSheetDialog
+          ref={feeDescriptionDialogRef}
+          height="fit-content"
+          containerStyle={{ paddingTop: 16 }}
+        >
```

Comment:
The next dialog uses `renderWhenOpen` prop, but this one doesn't. Should it be used here, too?

Reply:
The component doesn't have any innter logic so I thought this optimisation with renderWithOpen is not needed. Do we have a criteria when which case should be used?